src/database/models/client_profile_item.rs

```diff
@@ -0,0 +1,518 @@
+use std::path::PathBuf;
+
+use super::ids::*;
+use crate::database::models::DatabaseError;
+use crate::database::redis::RedisPool;
+use chrono::{DateTime, Utc};
+use dashmap::DashMap;
+use futures::TryStreamExt;
+use serde::{Deserialize, Serialize};
+
+// Hash and install path
+type Override = (String, PathBuf);
+
+pub const CLIENT_PROFILES_NAMESPACE: &str = "client_profiles";
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ClientProfile {
+    pub id: ClientProfileId,
+    pub name: String,
+    pub owner_id: UserId,
+    pub icon_url: Option<String>,
+    pub created: DateTime<Utc>,
+    pub updated: DateTime<Utc>,
+
+    pub game: ClientProfileGame,
+
+    pub loader_version: String,
+
+    pub maximum_users: i32,
+    pub users: Vec<UserId>,
+
+    // These represent the same loader
+    pub loader_id: LoaderId,
+    pub loader: String,
+
+    pub versions: Vec<VersionId>,
+    pub overrides: Vec<Override>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum ClientProfileGame {
+    Minecraft {
+        game_id: GameId,
+        game_name: String,
+        game_version_id: LoaderFieldEnumValueId,
+        game_version: String,
+    },
+    Unknown {
+        game_id: GameId,
+        game_name: String,
+    },
+}
```

Comment:
Do we need an enum here? Since we already have games dynamically in the database. Couldn't we just add a jsonb metadata field with game metadata (and then remove game_id and game_name fields

Reply:
Alternatively we could build this in with the loader fields system... but I don't think it's the best idea bc stuff like loader versions aren't on labrinth